internal/service/common/utils/repository.go

```diff
@@ -102,3 +103,33 @@ func Delete[T db.Model](ctx context.Context, db *pgxpool.Pool, uuid uuid.UUID) (
 
 	return result.RowsAffected(), nil
 }
+
+// Search retrieves a tuple from the database using arbitrary column values.  If no record is found an empty array is returned.
+func Search[T db.Model](ctx context.Context, db *pgxpool.Pool, expression bob.Expression) ([]T, error) {
+	// Build sql query
+	var record T
+	tags := GetAllDBTagsFromStruct(record)
+
+	query, args, err := psql.Select(
+		sm.Columns(tags.Columns()...),
+		sm.From(record.TableName()),
+		sm.Where(expression),
+	).Build()
+	if err != nil {
+		return []T{}, fmt.Errorf("failed to build query: %w", err)
+	}
+
+	// Run query
+	rows, _ := db.Query(ctx, query, args...) // note: err is passed on to Collect* func so we can ignore this
+	record, err = pgx.CollectExactlyOneRow(rows, pgx.RowToStructByNameLax[T])
+	if err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			slog.Info("No Entity found", "expression", expression, "table", record.TableName())
+			return []T{}, nil
+		}
+		return []T{}, fmt.Errorf("failed to call database: %w", err)
+	}
+
+	slog.Info("records found", "table", record.TableName(), "expression", expression)
```

Comment:
just realized that this log message and the ones for the functions above are misleading. Record can be an empty string.

Reply:
I feel as though returning an error when no result is found is a bit heavy handed.  Doing a select and having an empty result set is not an error.  It is simply an indication that there is no data or that the search criteria was too narrow.   This is how a normal "select * from table where ...;' works in any DB implementation.

Reply:
> just realized that this log message and the ones for the functions above are misleading. Record can be an empty string.

I've added a change to adjust some of the logs.   Not sure I completely understand your concern here though.

Reply:
The thing is that when I saw `Search` I started to think that `Find` is not ok. 

So, there are 2 functions from pgx, `CollectExactlyOneRow` which returns an error when the `SELECT` does not return rows and `CollectRows` which does not return an error when nothing is returned (I don't know why we are checking `pgx.ErrNoRows` here in `Search`and `FindAll`. Only returned by `CollectExactlyOneRow`),

For `Find`, we decided to use `CollectExactlyOneRow`. When no rows are returned, we ignore the error and return an empty slice. If there are no errors, that means that it is guaranteed to have a `Record`. However, we put it in a slice which forces us to check again outside if the slice returned by `Find` has length zero. We have to check if `Find` returns and error and then if the length of the slice is zero. And this is very odd to me. The verb `Find` should return an error when no rows are found, in my opinion.

But fine, if we insists on having this on `Find`, then we could use `CollectRows` instead of `CollectExactlyOneRow` because we are checking the length of the slice outside anyway. And if we do this, `Find` is very similar to `Search`. So, `Find` could define a `Quote` to search for the primary key and then call `Search` to avoid duplicating code.

Reply:
fixed